File: pt_038_bankovci_centi_20190913/bankovci_kovanci.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

import sys

usage="Usage:\n" + sys.argv[0] + " <amounts to calculate>"


if len(sys.argv) < 2:
    print(usage)
    exit(1)
    
eurs = []

for i in range(1, len(sys.argv)):
    # print(sys.argv[i])
    eurs.append(float(sys.argv[i].replace(",", ".")))


def bankovci_kovanci(eur):
    
    eur2 = round(eur, 2)
    
    print("===================================")
    print("*** Znesek: {} EUR ***".format(eur2))

    
    BC = {
    "Bankovec 100 EUR" : 0,
    "Bankovec 50 EUR" : 0,
    "Bankovec 20 EUR" : 0,
    "Bankovec 10 EUR" : 0,
    "Bankovec 5 EUR" : 0,
    "Kovanec 2 EUR" : 0,
    "Kovanec 1 EUR" : 0,
    "Kovanec 50 centov" : 0,
    "Kovanec 20 centov" : 0,
    "Kovanec 10 centov" : 0,
    "Kovanec 5 centov" : 0,
    "Kovanec 2 centa" : 0,
    "Kovanec 1 cent" : 0}
    
    rest1 = 0
    BC["Bankovec 100 EUR"] += int(eur2 / 100)
    rest1 += (eur % 100)
    
    rest2 = 0
    BC["Bankovec 50 EUR"] += int(rest1 / 50)
    rest2 += (rest1 % 50)

    rest3 = 0
    BC["Bankovec 20 EUR"] += int(rest2 / 20)
    rest3 += (rest2 % 20)
    
    rest4 = 0
    BC["Bankovec 10 EUR"] += int(rest3 / 10)
    rest4 += (rest3 % 10)
    
    rest5 = 0
    BC["Bankovec 5 EUR"] += int(rest4 / 5)
    rest5 += (rest4 % 5)
    
    rest6 = 0 
    BC["Kovanec 2 EUR"] += int(rest5 / 2)
    rest6 += (rest5 % 2)

    rest7 = 0 
    BC["Kovanec 1 EUR"] += int(rest6 / 1)
    rest7 += (rest6 % 1)
   
    rest7 = round(rest7 * 100, 0)
        
    rest8 = 0 
    BC["Kovanec 50 centov"] += int(rest7 / 50)
    rest8 += (rest7 % 50)
  
    rest9 = 0 
    BC["Kovanec 20 centov"] += int(rest8 / 20)
    rest9 += (rest8 % 20)
    
    rest10 = 0 
    BC["Kovanec 10 centov"] += int(rest9 / 10)
    rest10 += (rest9 % 10)
   
    rest11 = 0   
    BC["Kovanec 5 centov"] += int(rest10  / 5)
    rest11 += (rest10 % 5)
   
    rest12 = 0
    BC["Kovanec 2 centa"] += int(rest11 / 2)
    rest12 += (rest11 % 2)
   
    rest13 = 0
    BC["Kovanec 1 cent"] += int(rest12 / 1)
    rest13 += (rest12 % 1)
    
    for k, v in BC.items():
        if v != 0:
            print("{} = {}x".format(k, v))


for znesek in eurs:
    bankovci_kovanci(znesek)

File: pt_054_bankovci_centi_20190913/BankovciKovanci.java
import java.util.*;

public class BankovciKovanci {
    private Map<String, Integer> bc = new LinkedHashMap<String, Integer>();

    {   // initialization block
        bc.put("Bankovec 500 EUR", 0);
        bc.put("Bankovec 200 EUR", 0);
        bc.put("Bankovec 100 EUR", 0);
        bc.put("Bankovec 50 EUR", 0);
        bc.put("Bankovec 20 EUR", 0);
        bc.put("Bankovec 10 EUR", 0);
        bc.put("Bankovec 5 EUR", 0);
        bc.put("Kovanec 2 EUR", 0);
        bc.put("Kovanec 1 EUR", 0);
        bc.put("Kovanec 50 centov", 0);
        bc.put("Kovanec 20 centov", 0);
        bc.put("Kovanec 10 centov", 0);
        bc.put("Kovanec 5 centov", 0);
        bc.put("Kovanec 2 centa", 0);
        bc.put("Kovanec 1 cent", 0);
    }
    public static void main(String[] args) {
   

        List<Double> eurs = new ArrayList<Double>();
        for (String euro: args) {
            Double newEuro = Double.valueOf(euro);
            eurs.add(newEuro);
        }

        BankovciKovanci BC = new BankovciKovanci();

        for (Double znesek: eurs) {
            BC.getBaknkovciInKovanci(znesek);
        }
        
    } // end main


    public void getBaknkovciInKovanci(Double znesek) {
        System.out.println("===================================");
        System.out.printf("*** Znesek: %.2f EUR ***%n", znesek);

        double novirest1 = 0;
        bc.put("Bankovec 500 EUR", (int) (znesek / 500));
        novirest1 += (znesek % 500);
        
        double novirest2 = 0;
        bc.put("Bankovec 200 EUR", (int) (novirest1 / 200));
        novirest2 += (novirest1 % 200);
        
        double rest1 = 0;
        bc.put("Bankovec 100 EUR", (int) (novirest2 / 100));
        rest1 += (novirest2 % 100);
        
        double rest2 = 0;
        bc.put("Bankovec 50 EUR", (int) (rest1 / 50));
        rest2 += (rest1 % 50);

        double rest3 = 0;
        this.bc.put("Bankovec 20 EUR", (int) (rest2 / 20));
        rest3 += (rest2 % 20);

        double rest4 = 0;
        this.bc.put("Bankovec 10 EUR", (int) (rest3 / 10));
        rest4 += (rest3 % 10);

        double rest5 = 0;
        this.bc.put("Bankovec 5 EUR", (int) (rest4 / 5));
        rest5 += (rest4 % 5);

        double rest6 = 0; 
        this.bc.put("Kovanec 2 EUR", (int) (rest5 / 2));
        rest6 += (rest5 % 2);

        double rest7 = 0; 
        this.bc.put("Kovanec 1 EUR", (int) (rest6 / 1));
        rest7 += (rest6 % 1);

        rest7 = rest7 * 100;
       
        double rest8 = 0; 
        this.bc.put("Kovanec 50 centov", (int) (rest7 / 50));
        rest8 += (rest7 % 50);
      
        double rest9 = 0; 
        this.bc.put("Kovanec 20 centov", (int) (rest8 / 20));
        rest9 += (rest8 % 20);
        
        double rest10 = 0; 
        this.bc.put("Kovanec 10 centov", (int) (rest9 / 10));
        rest10 += (rest9 % 10);
       
        double rest11 = 0;   
        this.bc.put("Kovanec 5 centov", (int) (rest10  / 5));
        rest11 += (rest10 % 5);
       
        double rest12 = 0;
        this.bc.put("Kovanec 2 centa", (int) (rest11 / 2));
        rest12 += (rest11 % 2);
       
        double rest13 = 0;
        this.bc.put("Kovanec 1 cent", (int) (rest12 / 1));
        rest13 += (rest12 % 1);

        for(Map.Entry<String, Integer> entry: bc.entrySet()) {
            if (entry.getValue() != 0) {
                System.out.printf("%s = %dx%n",entry.getKey(), entry.getValue());
            }
        }

    }

} // end class